xds/internal/balancer/clusterresolver/resource_resolver.go

```diff
@@ -272,7 +287,9 @@ func (rr *resourceResolver) generateLocked() {
 }
 
 func (rr *resourceResolver) onUpdate() {
-	rr.mu.Lock()
-	rr.generateLocked()
-	rr.mu.Unlock()
+	rr.serializer.Schedule(func(context.Context) {
+		rr.mu.Lock()
+		rr.generateLocked()
+		rr.mu.Unlock()
+	})
```

Comment:
Something I just thought about with respect to this operation being async is this scenario. Say you have three priorities, and the first priority calls this. The CPU yields, and then two other updates corresponding to other priorities call this. Then, three generateLocked() calls, and three sends on the UpdateChannel with full configuration causing three priority JSON configurations to be sent downward. These are eaten here: https://github.com/grpc/grpc-go/blob/master/xds/internal/balancer/priority/balancer.go#L128, as no priorities would change. This seems minor and an ok performance hit, as this event is not per RPC and is triggered mainly by eds and dns resolver updates. But would love to know your thoughts.

Reply:
Yes, this seems like a super corner case and would make the `resourceResolver` unnecessarily complicated if we have to make it optimal, i.e send down updates to child policies only when absolutely required. As long as correctness is not affected, it shouldn't be an issue.

Reply:
Yup, no correct issue. Sounds good, seems minor. Similar in vein to discussions I've had with Doug about just sending redundant picker updates upward and not optimizing for not sending multiple of the same.